core/dbt/events/types.py

```diff
@@ -1719,6 +1713,42 @@ def message(self) -> str:
         return f"Command `{self.command}` {status} at {self.completed_at} after {self.elapsed:0.2f} seconds"
 
 
+class ShowNode(InfoLevel):
+    def code(self):
+        return "Q041"
+
+    def message(self) -> str:
+        if self.output_format == "json":
+            if self.is_inline:
+                return json.dumps({"show": json.loads(self.preview)}, indent=2)
+            else:
+                return json.dumps(
+                    {"node": self.node_name, "show": json.loads(self.preview)}, indent=2
+                )
+        else:
+            if self.is_inline:
+                return f"Previewing inline node:\n{self.preview}"
+            else:
+                return f"Previewing node '{self.node_name}':\n{self.preview}"
+
+
+class CompiledNode(InfoLevel):
+    def code(self):
+        return "Q042"
+
+    def message(self) -> str:
+        if self.output_format == "json":
+            if self.is_inline:
+                return json.dumps({"compiled": self.compiled}, indent=2)
+            else:
+                return json.dumps({"node": self.node_name, "compiled": self.compiled}, indent=2)
```

Comment:
[Not blocking] It makes sense to me that we'd want to support similar arguments for both `compile` and `show`, including `--output`. The current JSON output does feel a bit inconsistent between them:
```
10:54:04  {
  "node": "my_sql_model",
  "compiled": "select 1 as id"
}
...
10:54:17  Previewing node 'my_sql_model':
[{"id": 1.0}]
```
I don't have a very strong feeling about what to show here. The most important use case for JSON-formatted output is programmatic consumers of the `show` result set.